### PR TITLE
Parse output from composer and list missing extensions

### LINKED ISSUE #18: Check the necessary ext's are installed
zip, mbstring, etc

### PR DESCRIPTION
**Pull request by AydinHassan**
TODO: 
- [x] Tests

Fixes #18 

Features:
1. Adds required extensions to `composer.json` so this can only be global installed if required ext's are there.
2. validate command now checks if required extensions are installed.
3. Parses output from composer install and if there are missing extensions (while installing a workshop) we output a better message saying you need to install some extensions.

### CODE REVIEW

**Comment by AydinHassan** [comment]
Yo @mikeymike can you review please?

File: test/ComposerInstallerFactoryTest.php [hunk 1]
```diff
@@ -14,17 +14,23 @@
  */
 class ComposerInstallerFactoryTest extends PHPUnit_Framework_TestCase
 {
-    public function testCreate()
+    public function testTrue()
```
Comment (mikeymike) [inline test/ComposerInstallerFactoryTest.php]: mistake or coming back to it later?

File: test/ComposerInstallerTest.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        $output->setVerbosity(OutputInterface::VERBOSITY_VERBOSE);
+
+        $installer = new ComposerInstaller($input, $output, new Factory);
+        file_put_contents(sprintf('%s/composer.json', $this->tempDir), '{"name" : "learnyouphp", "require" : { "php": ">=5.6"}}');
+        $res = $installer->install($this->tempDir);
+
+        $this->assertFileExists(sprintf('%s/vendor', $this->tempDir));
+        $this->assertFileExists(sprintf('%s/composer.lock', $this->tempDir));
+
+        $expectedOutput  = "/Loading composer repositories with package information\n";
```
Comment (mikeymike) [inline test/ComposerInstallerTest.php:54]: Just a thought... are we not making this a fragile test by testing the output of another package? I see why it's done I'm just wondering if it's the best thing to do. 

Unfortunately I don't really have alternative other than checking that there was output which I guess in it self is a pretty weak/fragile test

File: test/IOFactoryTest.php [hunk 3]
```diff
@@ -14,40 +14,45 @@
  */
 class IOFactoryTest extends PHPUnit_Framework_TestCase
```
Comment (mikeymike) [inline test/IOFactoryTest.php]: safe to remove this test right?

**Comment by mikeymike** [comment]
@AydinHassan nice stuff dude... I like the approach with the IO much cleaner  👍

File: test/IOFactoryTest.php (hunk 3 above)
Comment (AydinHassan) [reply test/IOFactoryTest.php]: yep!

File: test/ComposerInstallerFactoryTest.php (hunk 1 above)
Comment (AydinHassan) [reply test/ComposerInstallerFactoryTest.php]: Yeah should be removed :D

File: src/InstallResult.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    private $missingExtensions;
+
+    /**
+     * @param string $exitCode
+     * @param string $output
+     */
+    public function __construct($exitCode, $output)
+    {
+        $this->exitCode = $exitCode;
```
Comment (AydinHassan) [inline src/InstallResult.php:33]: nice find @scrutinizer-auto-fixer !

File: test/ComposerInstallerTest.php (hunk 2 above)
Comment (AydinHassan) [reply test/ComposerInstallerTest.php:54]: it has no requires and it's not a real package - so it should be fine imo

**Pull request merged**